master/internal/db/postgres_trial.go

```diff
@@ -276,17 +276,17 @@ func TrialTaskIDsByTrialID(ctx context.Context, trialID int) ([]*model.RunTaskID
 	return ids, nil
 }
 
-// TrialByExperimentAndRequestID looks up a trial, returning an error if none exists.
-func TrialByExperimentAndRequestID(
-	ctx context.Context, experimentID int, requestID model.RequestID,
-) (*model.Trial, error) {
```

Comment:
Hm... are we guaranteed that requestIDs are not reused between experiments?

A long time ago, we generated UUIDs from the experiment seed, so two experiments with the same seed would actually get identical UUIDs.  Which sounds stupid, but was probably necessary in the earliest version of our searchers, which were built on a sort of writeahead log strategy with replays for restoring.

Based on the fact that model.NewRequestID still takes an RNG I'd say you probably still need the experiment id here, unfortunately.

Reply:
or... actually... is deleting all remaining NewRequestID's on your todo list?  It looks like it isn't used outside of tests, so maybe my feedback here is obsolete.

Reply:
although... you'd probably need to kill all non-terminal experiments to make this change, because right after upgrading, you are still going to have to restore experiments from where the request id wasn't unique.

Reply:
gah. yeah, i was afraid this was the case. added back experiment ID